pkg/router/template/template_helper.go

```diff
@@ -99,49 +104,19 @@ func genSubdomainWildcardRegexp(hostname, path string, exactPath bool) string {
 	return fmt.Sprintf(`^[^\.]*%s(|/.*)$`, expr)
 }
 
+// generateRouteRegexp is now legacy and around for backward
+// compatibility and allows old templates to continue running.
```

Comment:
why keeping this wrapper?

Reply:
For any existing custom haproxy templates to work, the helper function needs to exist. One scenario where this is needed is if you upgrade a router image but are still using the custom haproxy template file (based on an older template).